app/qan/mysql.go

```diff
@@ -353,7 +353,9 @@ func (h *MySQLMetricWriter) getQueryAndTables(class *event.Class) (query.QueryIn
 		bytes, _ := json.Marshal(query.Tables)
 		tables = string(bytes)
 	}
-
+	// We still want to store the fingerprint in the database
+	// even if an example is available
+	query.Query = class.Fingerprint
```

Comment:
`query.Query` gets inserted into the database as the `fingerprint`.

It uses `queryExample` to generate the abstract (which is good) but it should still store the fingerprint in the DB, otherwise, you end up with real literal values in the fingerprint and it's quite confusing in the display (especially since it uses the first example it ever sees as the fingerprint)

## Before Patch

![screen shot 2017-11-22 at 2 13 16 pm](https://user-images.githubusercontent.com/554927/33145694-70f6e056-cf8f-11e7-9478-c3be82206099.png)


## After Patch
![screen shot 2017-11-22 at 2 12 34 pm](https://user-images.githubusercontent.com/554927/33145675-5ab5659c-cf8f-11e7-80ec-00b71df07f73.png)

Reply:
Hi, 

Thanks for the clarification. 
Even when this seems to work, I would prefer to have clear variable names. Making query.Query to have the fingerprint is confusing and it would make harder to follow in the future.
Also, can we add a test case?

Reply:
> Even when this seems to work, I would prefer to have clear variable names. Making query.Query to have the fingerprint is confusing and it would make harder to follow in the future.

Should `query.Query` just be renamed to `query.Fingerprint` then? It seems like that variable should always be the fingerprint.

> Also, can we add a test case?
 
I don't have a lot experience with Go but I can give it a shot